test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1917,7 +1925,8 @@ protected Settings nodeSettings(int nodeOrdinal) {
             .put(SearchService.LOW_LEVEL_CANCELLATION_SETTING.getKey(), randomBoolean())
             .putList(DISCOVERY_SEED_HOSTS_SETTING.getKey()) // empty list disables a port scan for other nodes
             .putList(DISCOVERY_SEED_PROVIDERS_SETTING.getKey(), "file")
-            .put(featureFlagSettings());
+            .put(featureFlagSettings())
+            .put(customSettings);
```

Comment:
My understanding is this will not work since cluster gets created once for IT tests. I tested and verified as well. We will need to set these as dynamic cluster setting

```
SearchCancellationIT.testMSearchChildReqCancellationWithHybridTimeout [p0={}]

[2023-08-03T23:22:09,809][INFO ][o.o.s.q.QueryPhaseSearcherWrapper] [node_s1] Using default search over segments


SearchCancellationIT.testMSearchChildReqCancellationWithHybridTimeout [p0={"opensearch.experimental.feature.concurrent_segment_search.enabled":"true","search.concurrent_segment_search.enabled":"true"}]`

[2023-08-03T23:23:06,411][INFO ][o.o.s.q.QueryPhaseSearcherWrapper] [node_s0] Using default search over segments


```

Reply:
Specifically, it's the SUITE scope that causes the test cluster to get created once for all the tests within the test class:

```
@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE)
public class SearchCancellationIT extends OpenSearchIntegTestCase {
```

Reply:
@sohami @andrross the above works fine if we create a test specific cluster with the below annotation on the class.
So, in case of feature parameterization, we will have to change this annotation.
```
@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
public class SearchCancellationIT extends OpenSearchIntegTestCase {
```

Reply:
For now we can have mechanism for dynamic ones and if there is a need for non dynamic settings the move to `Scope.TEST` can be evaluated ? Individual tests can add that annotation`Scope.TEST` and can extend this new base class to provide the settings as parameter. This base class possibly can have the logic to check the annotation of derived class and use that to apply the settings properly (at node level or dynamic setting update).

Reply:
> This base class possibly can have the logic to check the annotation of derived class and use that to apply the settings properly (at node level or dynamic setting update).

I think we need this mechanism to work for both SUITE and TEST scope. I don't think it is viable to say that only TEST scope is supported for parameterized tests.

Reply:
The callout is we can only support dynamic settings update via parameterized mechanism for both SUITE and TEST scope clusters. Node level settings cannot be parameterized for SUITE scope clusters. So this mechanism will be used for dynamic settings only.